cli/src/args.rs

```diff
@@ -0,0 +1,475 @@
+use std::{collections::VecDeque, ffi::OsString, num::ParseIntError, path::PathBuf};
+
+use clap::{
+    builder::ValueParser, value_parser, Arg, ArgAction, ArgGroup, ArgMatches, Args, Command,
+    FromArgMatches, Parser, Subcommand, ValueEnum,
+};
+
+#[derive(Parser, Debug)]
+#[command(version, about)]
+pub struct ZipCli {
+    /// Write additional output to stderr.
+    #[arg(short, long)]
+    pub verbose: bool,
+    #[command(subcommand)]
+    pub command: ZipCommand,
+}
+
+#[derive(Subcommand, Debug)]
+pub enum ZipCommand {
+    /// do a compress
+    ///
+    /// Attributes can be set once and subsequently apply to all later entry arguments
+    /// until overridden.
+    Compress(Compress),
+    /// do an info
+    Info,
+    /// do an extract
+    Extract,
+}
+
+#[derive(ValueEnum, Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub enum CompressionMethodArg {
+    /// uncompressed
+    Stored,
+    /// with deflate (default)
+    Deflate,
+    /// with deflate64
+    Deflate64,
```

Comment:
Gate `Deflate64`, `Bzip2` and `Zstd` on their respective features.

Reply:
Done!